### ISSUE TITLE
IllegalStateException is thrown when validating invalid date value

### ISSUE THREAD

**Issue by GregorSz** [itplr-kosit/validator#41]
**Issue**
I just updated the validator to the [newest version](https://github.com/itplr-kosit/validator/releases/tag/v1.1.2) and discovered another behaviour in compare to [previous version](https://github.com/itplr-kosit/validator/releases/tag/v1.1.0) while validating an XRechnung which contain a not valid start date. (In release notes I can only find informations about bugfixes, not about changing the validation behaviour.)

**Scenario**
When validating an XRechnung with:
```
<cac:InvoicePeriod>
    <cbc:StartDate>Gestern</cbc:StartDate>
    <cbc:EndDate>2016-12-31+01:00</cbc:EndDate>
</cac:InvoicePeriod>
```

I get this exception:
```
Error evaluating (fn:exists(...) and fn:exists(...) and xs:date(EndDate) ge xs:date(StartDate) or fn:empty(...) or fn:empty(...)) in xsl:when/@test on line 2036 column 188 of EN16931-UBL-validation.xsl:
FORG0001: Invalid date "Gestern" (Non-numeric year component)
java.lang.IllegalStateException: Can not run schematron validation
```

... and validation is not successful. When validating the same XRechnung with version 1.1.0 of the validator no exceptions are thrown and a result object is returned with following schema violations:
```
cvc-datatype-valid.1.2.1: 'Gestern' ist kein gültiger Wert für 'date'.
cvc-complex-type.2.2: Element 'cbc:StartDate' darf kein Element [untergeordnete Elemente] haben, und der Wert muss gültig sein.
```

**Question**
I would expect that the result of the validation should be the same or at least no exception should be thrown. Can you please say if this behaviour has been intentionally changed or if it is just some kind of bug?

**Additional Information**
Versions in usage:
validationtool-1.1.2.jar
validator-configuration-xrechnung_1.2.2_2019-12-30.zip
Java(TM) SE Runtime Environment (build 1.8.0_241-b07)

**Comment by rkottmann** [comment]
Hi,

thx for reporting.

However, this is not issue of the validator itself. It is related to changed validation rules as configured in validator-configuration-xrechnung_1.2.2. Even more precisely `Error evaluating (fn:exists(...) and fn:exists(...) and xs:date(EndDate) ge xs:date(StartDate) or fn:empty(...) or fn:empty(...)) in xsl:when/@test on line 2036 column 188 of EN16931-UBL-validation.xsl:` tells me, it is a validation rule of https://github.com/ConnectingEurope/eInvoicing-EN16931 which we just integrate in our  validator-configuration-xrechnung.

I will try to aks them that they write this rule more robust and with better eroor messages.

**Comment by GregorSz** [comment]
Thank you for a quick response. 
Using the configuration version 1.2.2_2019-12-30 and validationtool-1.1.0 I don't get the exception. It occurs just after updating the validation tool to version 1.1.2.

**Comment by rkottmann** [comment]
Ah ok, then we might have two issues at the same time! We will investigate this too.

**Comment by rkottmann** [comment]
@apenski does it seem like a bug in 1.1.0 ?

**Comment by apenski** [comment]
the error report indicates, that the Version 1.1.0 is failing at schema validation step and stopping further processing while 1.1.2 is proceeding after this and then just fails on schematron step.

There was bug in 1.1.0 regarding malformed XML or garbage, that has been fixed in 1.1.1. Unfortunately this results in this error. I'll fix this

**Comment by GregorSz** [comment]
I assume that behvaiour of the version 1.1.0 is actually correct, is it true? Two schema violations make sense for me :) Moreover thank you for taking care of the problem. I appreciate it.

**Comment by apenski** [comment]
Yes, at least for your environment and test file 1.1.0 was working correctly. But your can easily have a scenario where schema validation works and schematron validation will fail with that same error in 1.1.0 too.

So these are bugs: 

1. a regression introduced in 1.1.1 that the schematron validation is started when schema validation fails

1. a bug in 1.1.x that processing errors while validating with schematron (and also when validation with schema) are not properly progagated to the callee via Result-object.

This is fixed in master and branch-1.1.x. I'll do some testing in the next few days before pushing a new release. Thanks for reporting!
